compute_tools/src/compute.rs

```diff
@@ -100,6 +91,45 @@ impl Default for ComputeState {
     }
 }
 
+#[derive(Clone, Debug)]
+pub struct ParsedSpec {
+    pub spec: ComputeSpec,
+    pub tenant: String,
+    pub timeline: String,
+    pub pageserver_connstr: String,
+    pub storage_auth_token: Option<String>,
+}
+
+impl TryFrom<ComputeSpec> for ParsedSpec {
```

Comment:
NIT: we don't parse anything really here, just validate a presence of a few strings, hence `ParsedSpec` name feels odd. Could be `SanitizedComputeSpec` or we could start parsing `tenant` and `timeline` values to justify the name 🙂